### ISSUE TITLE
[MINVOKER-156] Make it possible to have per project local repositories

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-invoker-plugin#459]
**[Martin Todorov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=carlspring)** opened **[MINVOKER-156](https://issues.apache.org/jira/browse/MINVOKER-156?redirect=false)** and commented

Currently, there is one local-repo which is used by all of the tests. In certain cases one may want to have or (for the same matter) not have a dependency in the local repository for a given test.

---

**Affects:** 1.8

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

The [Mock Repository Manager](http://mojo.codehaus.org/mrm/) should be able to help in this case. That way you have full control over the available dependencies

**Comment by jira-importer** [comment]
**[Martin Todorov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=carlspring)** commented

This is interesting! Could you please explain a bit about it?
I am working on https://github.com/carlspring/repository-unit-framework. This is basically a plugin which lights up a Jetty server with a "repository" restlet and you can mock artifacts in it and try to resolve or deploy them. How does the project you mentioned relate to this? (I had a look a it, but the pages are rather vague and contain little information).

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

This project was started by [stephenconnolly](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=stephenconnolly) in order to NOT depend on a remote repository and to have control over the available dependencies.
http://mojo.codehaus.org/mrm-maven-plugin/examples/invoker-tests.html explain how to integrate in with the maven-invoker-plugin.
In short: you'll need to specify a folder, which contains mainly pom-files, since the content of the jar/war/... is not interesting.
If a project needs one of these artifacts, the MRM will return an empty file. MRM uses Jetty as well.

Projects using the MRM:
* Maven Enforcer Plugin
* Siteskinner Maven Plugin
* JDiff Maven Plugin

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

This issue has been auto closed because it has been inactive for a long period of time. If you think this issue still applies, retest your problem with the most recent version of Maven and the affected component, reopen and post your results.
